lib/ansible/module_utils/facts/hardware/linux.py

```diff
@@ -209,6 +209,16 @@ def get_cpu_facts(self, collected_facts=None):
                     if 'vme' not in val:
                         xen_paravirt = True
 
+            if key == "flags":
+                if 'avx2' in val:
+                    cpu_facts['microarchitecture'] = 'x86_64v4'
+                elif 'avx' in val:
+                    cpu_facts['microarchitecture'] = 'x86_64v3'
+                elif 'popcnt' in val:
+                    cpu_facts['microarchitecture'] = 'x86_64v2'
+                elif 'mmx' in val:
+                    cpu_facts['microarchitecture'] = 'x86_64v1'
```

Comment:
I would probably just expose the flags, this is not as simple as it sounds as some stack others are mutually exclusive

Reply:
Ok.